servicetalk-http-api/src/main/java/io/servicetalk/http/api/HeaderUtils.java

```diff
@@ -557,14 +557,33 @@ protected final void initNext(CharSequence cookieHeaderValue) {
          */
         private HttpCookiePair findNext(CharSequence cookieHeaderValue) {
             int semiIndex = nextCookieDelimiter(cookieHeaderValue, nextNextStart);
-            HttpCookiePair next = DefaultHttpCookiePair.parseCookiePair(cookieHeaderValue, nextNextStart, semiIndex);
+            int nameLength = indexOf(cookieHeaderValue, '=', nextNextStart) - nextNextStart;
```

Comment:
the previous `parseCookiePair` method is no longer used, remove it?

should the `nameLength` validation be done in the `parseCookiePair` method?